src/network_adapter.hpp

```diff
@@ -101,54 +101,36 @@ class NetworkAdapter final : public dbus::FindObjectDBusQuery
         return criteria;
     }
 
-    static const DbusVariantType
-        formatEnabledField(const PropertyName& currentProp,
-                           const DbusVariantType& value,
-                           DBusInstancePtr instance)
+    void setEnabled(DBusInstancePtr& instance) const
     {
-        static const std::map<PropertyName, app::entity::MemberName>
-            buildPropRelation{
-                {propPresent, fieldFunctional},
-                {propFunctional, fieldPresent},
-            };
+        // default enabled
+        bool enabled = true;
         try
         {
-            auto targetRelatedFieldIt = buildPropRelation.find(currentProp);
-            if (targetRelatedFieldIt == buildPropRelation.end())
-            {
-                throw std::logic_error(
-                    "Can't find related network status field");
-            }
-
-            bool targetFieldRemote =
-                instance->getField(targetRelatedFieldIt->second)
-                    ->getBoolValue();
+            bool functional =
+                instance->getField(fieldFunctional)->getBoolValue();
+            bool present = instance->getField(fieldPresent)->getBoolValue();
```

Comment:
Hm... What will happen with an object that has no `functional` and `present=false`?

Reply:
Saying `separate interfaces might incoming by separate instance-objects` maybe I dropped the ball a little bit. To be completely honest the Instance object will be same once class-instance for each call (at the real-dbus object border), but  the `supplementByStaticFields` will be called on each dbus-interface obtaining. So, summary we will acquire the full initialized instance-object at end of dbus-calling each appropriate dbusInstance interface and the last call of `supplementByStaticFields` for the Instance will have both  fields `functional` and `present` always.

**Remark:** the new incoming implementation, that will be published soon, will haven't that shortage.

Reply:
There is a too complicated architecture and I'm lazy to dip into. :)

On these lines I see code that can throw an exception about the nonexistent field or return some hardcoded default value from `instanceNotFound()`

- With the exception about `functional` we lose the attempt to read `presence`.
- With `instanceNotFound()` I don't see what exactly that default value is. I assume that non existent `functional` should be interpreted as `true`. But will this code do so?

Reply:
I guess you means the `getField()` method and that is no entirely true.
The `instanceNotFound` will thrown if the Entity::Member not define. The [line 133](https://github.com/YADRO-KNS/obmc-yadro-webapp/pull/12/files/cacf64f9ad36a5177ba0c84b4453e905aabe3bb9#diff-24354e42db1c89ad5d898927c868f5c64f1e9bea2ee369779718cb7e5ed3bc0eR133) defines both fieldFunctional and fieldPresent. So, in the IEntity that fields will founds always.
The another side is that fields might be not populated(will give "N/A" value) at some time point. And the logic of that cases will described above.